x-pack/plugins/licensing/server/types.ts

```diff
@@ -44,17 +44,23 @@ export interface RawLicense {
  * The APIs exposed on the `licensing` key of {@link RequestHandlerContext} for plugins that depend on licensing.
  * @public
  */
-export interface LicensingRequestHandlerContext {
+export interface LicensingApiRequestHandlerContext {
   featureUsage: FeatureUsageServiceStart;
   license: ILicense;
 }
 
-declare module 'src/core/server' {
-  interface RequestHandlerContext {
-    licensing: LicensingRequestHandlerContext;
-  }
+/**
+ * @internal
+ */
+export interface LicensingRequestHandlerContext extends RequestHandlerContext {
+  licensing: LicensingApiRequestHandlerContext;
```

Comment:
@elastic/kibana-core we can go further and create a convention for every plugin to make this interface public. In this way, the dependent plugins can leverage them to infer the final type for local `RequestHandlerContext` version:
```typescript
// another plugin
import type { LicensingRequestHandlerContext } from '../licensing/server';
export interface MyRequestHandlerContext extends RequestHandlerContext, LicensingRequestHandlerContext {
}
```
The drawbacks of the approach:
- we don't do this for plugin contract types
- requires complex declarations for optional plugins:
```typescript
// another plugin
import type { LicensingRequestHandlerContext } from '../licensing/server';
export type MyRequestHandlerContext = RequestHandlerContext & Partial<LicensingRequestHandlerContext>
```